### ISSUE TITLE
VS Debugger doesn't work with 'system' modules

### ISSUE THREAD

**Issue by henrify** [Microsoft/TypeScript#8347]
I'm trying to get basic hello world working with 'system' modules, but no matter what, the breakpoints are not hit.

Steps:
- Select 'system' modules, 'combine JavaScript output into file' and 'generate source maps' in TypeScript Build options
- Build with VS2013.5 Ultimate, TS 1.8.5
- Set breakpoint in TS source
- Start debugging application which loads the output file with SystemJS

Expected:
- Breakpoint hits, suspends execution and allows stepping through TS project source

Actual:
- The module loads and executes correctly
- Successful HTTP request is made to load the source map
- The breakpoint set in TS source is not hit
- Warning 'Breakpoint will not be currently hit. Sourcemaps have not been loaded' is displayed next to the breakpoint.
- The executing TS shows as 'eval code [dynamic]' in visual studio with raw JS content

Variants:
- I've tried various combinations of relative and absolute source map and TS file root directories in the TypeScript build options, no change.

Repro:
- The full VS test solution is attached

[SystemJsTest.zip](https://github.com/Microsoft/TypeScript/files/239632/SystemJsTest.zip)

**Comment by mhegazy** [comment]
from a quick look it seems that the loader is wrapping the generated code in an additoinal function..

app.js as generated by the compiler looks like:

```js
System.register("app", [], function(exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    function hello() {
        console.log("world");
    }
    exports_1("hello", hello);
    return {
        setters:[],
        execute: function() {
        }
    }
});
//# sourceMappingURL=app.js.map
```

the one loaded in the browser is:

```js
(function(System, SystemJS) {System.register("app", [], function(exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    function hello() {
        console.log("world");
    }
    exports_1("hello", hello);
    return {
        setters:[],
        execute: function() {
        }
    }
});
//# sourceMappingURL=app.js.map
})(System, System);
```

@guybedford any insight here?

**Comment by mhegazy** [comment]
looking further. having code after the source map reference is not in the sourcemap spec. so the debugger never loads the sourcemap and considers this reference invalid. but it seems that browsers honor this anyways. we can relax the check on the VS debugger side and attempt to load the map at least.

**Comment by sheetalkamat** [comment]
This is resulting because our debugger plugin doesn't support ```@sourceUrl``` syntax yet. I am working on it and will let you know when it is complete.

**Comment by lelezeus88** [comment]
I've noticed that with version 0.19.18 of SystemJs the browser correctly loads the ts and js file, but from version 0.19.19 these files are no more loaded and apperars an "eval code" file.
I also noticed in SystemJs' release note of 0.19.19  this fix : "Source maps sanitization adjustments and comprehensive support for plugin / transpiler sourcemap chaining".
Maybe it has nothing to do with this problem, I hope it helps.

**Comment by kabua** [comment]
kabua/systemjs#450

**Comment by kabua** [comment]
I believe this issue is also related to this issue: https://github.com/systemjs/systemjs/issues/450

**Comment by sheetalkamat** [comment]
The reported scenario didn't work for two things. 
Debugger didn't support @sourceUrl syntax as well it didn't work with eval blocks.
I have just made changes to debugger extension to support these two things and your solution should work after you install the next release version of typescript that ships the debugger changes.

**Comment by henrify** [comment]
@sheetalkamat That's fantastic, thank you!

Can you advice what is "next release version" (major version or point release), and when it is expected?

**Comment by kabua** [comment]
@sheetalkamat, thank you! 

Also, would you be kind enough to post it (here) when these changes have made it in the "next release version"? Please include the full version number (and a link would be nice), Thanks.

**Comment by mhegazy** [comment]
this should be available in the next release (Typescript 2.0). hopefully we should have a pre-release within the next few weeks.

**Comment by jonaskello** [comment]
I tried to verify this fix using the nightly drops. Using the example project provided in the zip, I installed nuget Microsoft.TypScript.MSBuild 1.9.0-beta-20160613-2 and npm install 1.9.0-dev.20160614-1.0. I also ran the powershell script to update the language service for VS2015. However the breakpoint is still not hit and I get this message in the VS2015 output pane:
```Unsupported format of the sourcemap.Unsupported format of the sourcemap.```
